### DISCUSSION TITLE
M106 Error Ender 3v3

### DISCUSSION THREAD

**Discussion by bhast2** [Guilouz/Creality-Helper-Script-Wiki#561] [Issues]
I am getting this error on M106 macro

`!! {"code":"key165", "msg": "Error evaluating 'gcode_macro M106:gcode': jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'S'
`

**Comment by Guilouz** [comment]
There is no change in M106 macro for Ender-3 V3 in my script.

**Reply by evolventum** [reply]
there is problem with gcode_macro.cfg - after changing the **variable_fan0_min** from 140 to 0 system will be correctly calculate M106 macro with real range from 0 to 255, and when variable was 140 the commands in terminal _SET_PIN PIN=fan0 VALUE=100_ and _SET_PIN PIN=fan0 VALUE=100_ did gave different RPM on fan, so the g-code also did gave incorrect fan speed when printing
